unit_tests/README.md

```diff
@@ -0,0 +1,63 @@
+# DTS scripts unit tests
+
+This directory contains stub scripts for testing DTS update and deploy logic
+in complex scenarios.
+
+There are some unsolved dependencies like dasharo_ectool, cbfstool, futility.
+
+## Test switching to heads and back with updates
+
+```bash
+export BOARD_VENDOR="Notebook" SYSTEM_MODEL="NV4xPZ" BOARD_MODEL="NV4xPZ"
+```
+
+1. Dasharo v1.7.2 on NV4x_PZ eligible for updates to heads with heads DES and
+   regular update:
+
+    ```bash
+    export BIOS_VERSION="Dasharo (coreboot+UEFI) v1.7.2" TEST_DES=y && source dts-boot
```

Comment:
For me, it ends with: `dts-boot:3: = not found`. Should I just execute these two commands?

Reply:
RIght now, basically yes.

1. Set the board you wish to test: `export BOARD_VENDOR="Notebook" SYSTEM_MODEL="NV4xPZ" BOARD_MODEL="NV4xPZ"`
2. Set the current version you have "installed" and if you have DES or not and run:
`export BIOS_VERSION="Dasharo (coreboot+UEFI) v1.7.2" TEST_DES=y && source dts-boot`

Reply:
@miczyg1 I also did that and got similar results as @macpijan 

```
λ export BOARD_VENDOR="Notebook" SYSTEM_MODEL="NV4xPZ" BOARD_MODEL="NV4xPZ"
λ export BIOS_VERSION="Dasharo (coreboot+UEFI) v1.7.2" TEST_DES=y && source dts-boot
dts-boot:3: = not found
```

Reply:
Okay, I will take a look

Reply:
This fixed it for me: https://github.com/Dasharo/meta-dts/pull/95/commits/74a271545271c8e7392802e8fbfc31c8e264ac9c

It is important to run those commands from unit_tests directory

Reply:
> It is important to run those commands from unit_tests directory

@miczyg1 I think that more important is to have those utilities you mentioned in `Prerequisities` and we do not have instructions how to do that.

Maybe it would be better to add a script to run DTS in QEMU and then execute this unit_tests there?

Reply:
@TomaszAIR that approach doesn't make much sense. The whole idea of these unit tests was to test the script logic without building whole distro. It is much faster and has the potential to be run in CI easily.

Reply:
By unit testing definition:

```
In computer programming, unit testing is a software testing method by which individual units of source code—sets of one or more computer program modules together with associated control data, usage procedures, and operating procedures—are tested to determine whether they are fit for use.[1] It is a standard step in development and implementation approaches such as Agile.
```

And unit definition:

```
Unit would be the smallest component that can be isolated within the complex structure of an app. It could be a function, a subroutine, a method or property.
```

Of course, whether the whole script logic is the smallest component may be debatable.

Reply:
@miczyg1 Ok, I think I've discovered what was the main reason why @macpijan nor I were not able to run these tests. Probably we were running them from a different shell (zsh vs bash) which ended up in reported errors. I do not reproduce this when dts-boot script has added exec permissions. So I suggest to change it (done in https://github.com/Dasharo/meta-dts/pull/95/commits/470addbaff75d49ba2d8a90ed23a07f3137f60a0) and modify a little README from unit_tests (done in https://github.com/Dasharo/meta-dts/pull/95/commits/b3b835f51a63fc42439b18dd72911bf658fe51a9). In addition, I changed the commands for CMD_POWEROFF and CMD_REBOOT (done in https://github.com/Dasharo/meta-dts/pull/95/commits/470addbaff75d49ba2d8a90ed23a07f3137f60a0#diff-efc848d5fa8283a79f5d2856ae5e06dcc2deede001121144d5b8626d4f6aad98L55) to `exit 0` to be able to terminate the execution of the script (without this I was circling in the main loop).

@miczyg1 let me know if you are ok with this.

Reply:
I suggest:

```
CMD_POWEROFF="echo poweroff && exit 0"
CMD_REBOOT="echo reboot && exit 0"
```

That way we know what is going on. Created separate thread for that. Closing this one as you have found the solution.